cylc/flow/config.py

```diff
@@ -2265,6 +2266,17 @@ def load_graph(self):
             self.workflow_polling_tasks.update(
                 parser.workflow_state_polling_tasks)
             self._proc_triggers(parser, seq, task_triggers)
+            self.check_outputs(
+                [
+                    task_output
+                    for task_output in parser.task_output_opt
+                    if task_output[0]
+                    in [
+                        task_output.split(':')[0]
+                        for task_output in parser.terminals
+                    ]
+                ]
```

Comment:
I think this needs to check every terminal output, but it's actually checking every output (both terminal and non-terminal) of any task that has at least one terminal output. So there's some unnecessary checking going on.

Reply:
(Also, that comprehension would probably be clearer if lines 2275-6 used a different name for `task_output`, as it's a different variable from that in lines 2271-3).

Reply:
I'll see if I can work my way around this, but ther reason I need to use every output is that terminals only carries the raw data. If the user sets `foo:fail` this is not the same as `foo:failed`.